### ISSUE TITLE
Pyodide Worker

### ISSUE THREAD

**Issue by stefnotch** [mgreminger/EngineeringPaper#2]
Regarding Pyodide, EngineeringPaper currently has
- A custom version of Pyodide, which is great and also a fair bit sleeker than normal Pyodide
- Pyodide running in the main thread

I think that it could be improved by 
- [moving Pyodide to a web worker](https://github.com/stefnotch/quantum-sheet/tree/master/public) 
- [using a shared worker in development mode](https://github.com/stefnotch/quantum-sheet/blob/2ff8ae6439eea8c2fdc67c280a360764e22a1dc9/src/cas/pyodide-cas.ts#L164-L173) and then opening a page in a new tab that [simply keeps the worker alive](https://github.com/stefnotch/quantum-sheet/blob/master/public/pyodide-worker-keep-alive.html)
- [(optionally letting it survive hot-reloading)](https://github.com/stefnotch/quantum-sheet/blob/master/src/cas/pyodide-cas.ts#L215-L218)

Are there any other possible improvements?

Also, since I'm struggling to building Pyodide myself, I wanted to ask you if this would be something we could both work on?

**Comment by mgreminger** [comment]
Yes, Pyodide will eventually need to be moved to a web worker. The calculations are pretty fast now and the delay doesn't impact usability but it will eventually become an issue when solving bigger problems.

The only difference between standard Pyodide and what I'm using is the addition of the antlr4 runtime and the development version of Sympy. Pyodide itself is unaltered. There are simpler ways to include these packages then by creating a Pyodide build environment since both packages are pure Python, but I haven't made those changes yet.
